presto-hive/src/main/java/com/facebook/presto/hive/CachingDirectoryLister.java

```diff
@@ -49,25 +49,28 @@
         implements DirectoryLister
 {
     private final Cache<Path, List<LocatedFileStatus>> cache;
-    private final Set<SchemaTableName> tableNames;
+    private final Set<SchemaTableName> cachedTableNames;
 
     @Inject
     public CachingDirectoryLister(HiveClientConfig hiveClientConfig)
     {
-        this(hiveClientConfig.getFileStatusCacheExpireAfterWrite(), hiveClientConfig.getFileStatusCacheMaxSize(), hiveClientConfig.getFileStatusCacheTables());
+        this(
+                hiveClientConfig.getFileStatusCacheExpireAfterWrite(),
+                hiveClientConfig.getFileStatusCacheMaxSize(),
+                hiveClientConfig.getFileStatusCacheTables().stream()
+                        .map(CachingDirectoryLister::parseTableName)
+                        .collect(Collectors.toSet()));
     }
 
-    public CachingDirectoryLister(Duration expireAfterWrite, long maxSize, List<String> tables)
+    public CachingDirectoryLister(Duration expireAfterWrite, long maxSize, Set<SchemaTableName> tables)
     {
-        this.cache = CacheBuilder.newBuilder()
+        cache = CacheBuilder.newBuilder()
                 .maximumWeight(maxSize)
                 .weigher((Weigher<Path, List<LocatedFileStatus>>) (key, value) -> value.size())
                 .expireAfterWrite(expireAfterWrite.toMillis(), TimeUnit.MILLISECONDS)
                 .recordStats()
                 .build();
-        this.tableNames = tables.stream()
-                .map(CachingDirectoryLister::parseTableName)
-                .collect(Collectors.toSet());
+        cachedTableNames = tables;
```

Comment:
nit: Maybe `cachedTableNames = ImmutableSet.copyOf(requireNonNull(cachedTableNames, "cachedTableNames is null")`. I'm not sure if a defensive copy is necessary here though.